<a href="https://colab.research.google.com/github/GaabrielCoosta/SoulCodeAcademy/blob/main/Projeto_01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Dicionário da base de dados


Atualização da biblioteca gspread

In [ ]:
!pip install --upgrade gspread

1- Autenticando o GoogleSheets, 2- Vinculando a tabela Dicionario_rh, 3 - Exibindo a tabela vinculada

In [101]:
from google.colab import auth
auth.authenticate_user()

import gspread
from google.auth import default
creds, _ = default()

gc = gspread.authorize(creds)

dic = gc.open_by_key('11qpux1tUnbzs3udV8JY6KMgzUfdBkntM_an6O9D_yoc')
ws = dic.worksheet('dicionariorh')
# get_all_values gives a list of rows.
rows = ws.get_all_values()
#print(rows)

# Convert to a DataFrame and render.
import pandas as pd
dicionario = pd.DataFrame.from_records(rows)

display(dicionario)

0  \
0                                 Nome do Atributo   
1                                            Idade   
2                                         Desgaste   
3                               Viagem de negócios   
4                                Diárias recebidas   
5                                     Departamento   
6                                Distância de Casa   
7                                     Escolaridade   
8                                 Área de Formação   
9       Número de contratos de trabalho na empresa   
10                                  ID Funcionário   
11  Nível de Satisfação com o ambiente de trabalho   
12                                          Gênero   
13                                Salário por hora   
14            Nível de envolvimento com o trabalho   
15                               Nível hierárquico   
16                                           Cargo   
17                       Satisfação com o trabalho   
18                                    Estado civil   
19                                    Renda mensal   
20                                  Salário Mensal   
21               qtde de empresas que já trabalhou   
22                                  Maior de idade   
23                                  Faz hora extra   
24                percentual de aumento de salário   
25                           score de performance    
26             Satisfação nas relações no trabalho   
27                      Jornada padrão de trabalho   
28          opção de remaunração variável em ações   
29                       Total de anos trabalhados   
30   Qtde de treinamentos realizados no último ano   
31                        Equilíbrio vida trabalho   
32            Total de anos trabalhados na empresa   
33                Anos trabalhados na função atual   
34                    Anos desde a última promoção   
35             Anos trabalhando com o atual gestor   

                                                    1  \
0                                         Significado   
1                                               Idade   
2   Pediu demissão (SIM - saiu da empresa) (NÃO - ...   
3     Frequência que o funcionário viaja pela empresa   
4           Diárias recebidas pelos funcionários (R$)   
5                     Departamento que está vinculado   
6             Distância de casa para o trabalho em km   
7                 Nível de escolarida do funcionários   
8                    Área de formação do funcionários   
9   Quantas vezes o funcionário é contado para o t...   
10                  Número de registro do funcionário   
11  Nível de satisfação do funcionário com o ambie...   
12                             Gênero do Funcionário    
13              Salário por hora do funcionário em R$   
14  Nível de envolvimento exigido do funcionário p...   
15  Nível hierárquico (Quanto maior mais important...   
16                    Cargo do funcionário na empresa   
17                 Nível de Satisfação com o trabalho   
18                        Estado civil do funcionário   
19  Renda mensal (quanto de dinheiro líquido entra...   
20                               Salário mensal bruto   
21  Quantidade de empresas que o funcionário já tr...   
22                    O Funcionário É Maior de idade?   
23                      O funcionário faz hora extra?   
24  Percentual de aumento no salário que o funcion...   
25  Pontuação de performance do funcionário na emp...   
26  Nível de satisfação do funcionário com suas re...   
27            Jornada de trabalho em horas quinzenais   
28  Nível de remuneração variável por opção de bon...   
29  Número total de anos trabalhados pelo funcionário   
30  Quantidade de treinamentos realizados pelo fun...   
31  Percepção do funcionário do equilíbrio vida/tr...   
32  Número total de anos trabalhados pelo funcioná...   
33           Número de anos trabalhado no cargo atual   
34  Número de anos desde a última promoção do func...  

# Base de dados dos funcionários que pediram demissão por desgaste


In [47]:
desgaste = gc.open_by_key('15aKHQt4uVzhAs35NwcL_8G3IRs2hzzvXuYAmZZzxhUk')
ws = desgaste.worksheet('basedesgaste')
# get_all_values gives a list of rows.
rows = ws.get_all_values()
#print(rows)

# Convert to a DataFrame and render.
import pandas as pd
base_desgaste = pd.DataFrame.from_records(rows)

base_desgaste.head()

0             1       2                                  3   \
0  Idade  Estado Civil  Gênero  qtde de empresas que já trabalhou   
1     41             0       0                                  8   
2     37             0       1                                  6   
3     28             0       1                                  5   
4     36             0       1                                  7   

                          4                                     5   \
0  Total de anos trabalhados  Total de anos trabalhados na empresa   
1                          8                                     6   
2                          7                                     0   
3                          6                                     4   
4                         10                                     5   

                          6                                               7   \
0  Satisfação com o trabalho  Nível de Satisfação com o ambiente de trabalho   
1                          4                                               2   
2                          3                                               4   
3                          3                                               3   
4                          1                                               3   

                                    8                                     9   \
0  Satisfação nas relações no trabalho  Nível de envolvimento com o trabalho   
1                                    1                                     3   
2                                    2                                     2   
3                                    2                                     2   
4                                    2                                     2   

   ...                                12                                   13  \
0  ...  Anos trabalhados na função atual  Anos trabalhando com o atual gestor   
1  ...                                 4                                    5   
2  ...                                 0                                    0   
3  ...                                 2                                    3   
4  ...                                 3                                    3   

                             14  \
0  Anos desde a última promoção   
1                             0   
2                             0   
3                             0   
4                             0   

                                              15                    16  \
0  Qtde de treinamentos realizados no último ano  score de performance   
1                                              0                     3   
2                                              3                     3   
3                                              4                     3   
4                                              4                     4   

                                 17                  18                   19  \
0  percentual de aumento de salário   Salário por hora    Diárias recebidas    
1                                11          R$  94,00         R$  1.102,00    
2                                15          R$  92,00         R$  1.373,00    
3                                14          R$  93,00           R$  103,00    
4                                23          R$  82,00         R$  1.218,00    

                       20                      21  
0   Salário Mensal BRUTO    Renda mensal líquida   
1          R$  19.479,00            R$  5.993,00   
2           R$  2.396,00            R$  2.090,00   
3          R$  12.947,00            R$  2.028,00   
4           R$  6.986,00            R$  3.407,00   

[5 rows x 22 columns]

# Estatística Descritiva e Dispersão
# Medidas descritivas
- A estatística descritiva organiza e descreve os dados de três maneiras: por meio de tabelas, de gráficos e de medidas descritivas.

# Medidas de dispersão
- A medidas de dispersão nos ajudam a entender como os dados variam em torno da média, ou seja, também ajudam a descrever a tendência nos dados.

Descrições das variáveis:

1.   Medidas de Tendência Central

  *   md = Média
  *   mediana = Mediana
  *   moda = Moda

2.   Medidas de Dispersão

  *   dp = Desvio Padrão
  *   va = Variância
  *   q1 = Quartil 1
  *   q3 = Quartil 3
  *   min = Mínimo valor
  *   max = Máximo valor
  *   amp = Amplitude Total
  *   aiq = Amplitude Interquatilica




In [100]:
# carregando os dados da tabela filtrada dos funciónarios que pediram demissão

desgastados = pd.read_csv("/content/drive/MyDrive/Projeto01-Colab_GitHub/Baserh_Desgastados  - basedesgaste (1).csv")
#display(desgastados)


# *********** Medidas de Tendencia Central ************

# Média
md = desgastados.mean()
#print("A média dos atributos são:")
#display(md)

# Mediana
mediana = desgastados.median()
#print("A mediana dos atributos são:")
#display(mediana)

# Moda
moda = desgastados.mode()
#print("A moda dos atributos são:")
#display(moda)


# *********** Medidass de dispersão *************
# Máximo
#print("O máximo dos atributos são:")
max = desgastados.max()
#display(max)

# Mínimo
min = desgastados.min()
#print("O mínimo dos atributos são:")
#display(min)

# Desvio Padrão
dp = desgastados.std()
#print("O desvio padrão dos atributos são:")
#display(dp)

# Variância
va = desgastados.var()
#print("A variância dos atributos são:")
#display(va)

# Amplitude Total
amp = desgastados.max() - desgastados.min()
#print("A amplitude total dos atributos são:")
#display(amp)

# Quartil 1
q1 = desgastados.quantile(.025)
#print("O quartil 1 dos atributos são:")
#display(md)

# Quartil 3
q3 = desgastados.quantile(0.75)
#print("O quartil 3 dos atributos são:")
#display(md)

# Amplitude Interquartil
aiq = desgastados.quantile(.75) - desgastados.quantile(.25)
#print("A amplitude interqualil dos atributos:")
#display(aiq)



Idade  Estado Civil  Gênero  qtde de empresas que já trabalhou  \
0       41             0       0                                  8   
1       37             0       1                                  6   
2       28             0       1                                  5   
3       36             0       1                                  7   
4       34             0       1                                  2   
..     ...           ...     ...                                ...   
232     23             1       1                                  1   
233     29             1       1                                  9   
234     56             1       1                                  8   
235     50             2       1                                  7   
236     50             2       1                                  4   

     Total de anos trabalhados  Total de anos trabalhados na empresa  \
0                            8                                     6   
1                            7                                     0   
2                            6                                     4   
3                           10                                     5   
4                            8                                     4   
..                         ...                                   ...   
232                          1                                     1   
233                          4                                     2   
234                         14                                    10   
235                         12                                     6   
236                         20                                     3   

     Satisfação com o trabalho  \
0                            4   
1                            3   
2                            3   
3                            1   
4                            1   
..                         ...   
232                          1   
233                          4   
234                          3   
235                          3   
236                          1   

     Nível de Satisfação com o ambiente de trabalho  \
0                                                 2   
1                                                 4   
2                                                 3   
3                                                 3   
4                                                 2   
..                                              ...   
232                                               4   
233                                               1   
234                                               4   
235                                               2   
236                                               4   

     Satisfação nas relações no trabalho  \
0                                      1   
1                                      2   
2                                      2   
3                                      2   
4                                      3   
..                                   ...   
232                                    1   
233                                    2   
234                                    4   
235                                    4   
236                                    2   

     Nível de envolvimento com o trabalho  ...  \
0                                       3  ...   
1                                       2  ...   
2                                       2  ...   
3                                       2  ...   
4                                       3  ...   
..                                    ...  ...   
232                                     3  ...   
233                                     3  ...   
234                                     3  ...   
235                                     3  ...   
236                                     2  ...   

     Anos trabalhados na função atual  Anos trabalhando com o atual gestor  \
0        

# Distribuições de Frequência


#Frequência de demissões por nível hierarquico


In [129]:
nh = gc.open_by_key('1_PgwcyaShF8dn4AbFLTOZsQLswJfqUNxpxFROZqkHac')
ws = nh.worksheet('nivelhierarquico')
# get_all_values gives a list of rows.
rows = ws.get_all_values()
#print(rows)

# Convert to a DataFrame and render.
import pandas as pd
nivel = pd.DataFrame.from_records(rows)
print(nivel)

       0           1                2             3                  4
0  Nivel  Quantidade  Quantidade acum  % Quantidade  % Quantidade acum
1                143              143        60,34%             60,34%
2                 52              195        21,94%             82,28%
3                 32              227        13,50%             95,78%
4                  5              232         2,11%             97,89%
5                  5              237         2,11%            100,00%
6  Total         237                        100,00%                   


- Ao total, **237** funcionários pediram demissão. **60,34%** possuiam **nivel 1** e **21,94% nível 2**, sendo eles os níveis mais baixos de uma escala de **1 à 5**. A soma dos dois níveis representam **82,28%** dos funcionários desgastados.

#Frequência de demissões por anos de vida

In [141]:
idade = gc.open_by_key('1SeUJVkmxckdtLG_o-oX1YpRhajd5KJO8-ZqD5xD-RGc')
ida = idade.worksheet('idade')
# get_all_values gives a list of rows.
rows = ida.get_all_values()
#print(rows)

# Convert to a DataFrame and render.
import pandas as pd
idaded = pd.DataFrame.from_records(rows[1:7])
print(idaded)

         0           1                2             3                  4
0    Idade  Quantidade  Quantidade acum  % Quantidade  % Quantidade acum
1  18 - 25          44               44        18,57%             18,57%
2  26 - 36         122              166        51,48%             70,04%
3  37 - 45          37              203        15,61%             85,65%
4  46 - 58          34              237        14,35%            100,00%
5    Total         237                        100,00%                   


- Através dessa análise obtivemos o dado que **70,04%** dos funcionários desgastados possuíam entre **18 e 36** anos, e que **51,48%** possuíam entre **26** e **36** anos de idade.

#Frequência de demissões por gênero